frame/vault/src/tests.rs

```diff
@@ -525,93 +525,125 @@ proptest! {
 proptest! {
 	#![proptest_config(ProptestConfig::with_cases(100))]
 
+	// The strategy withdraw is calculated multiplying the deposited value by
+	// (accounts.len() % 100) and then the strategy balance representing a loss is calculated
+	// using the same percentage meaning that `deposit` > `withdraw` > `strategy balance`.
+	//
+	// `strategy_moment` is when the strategy generates profits or losses. Before or after that
+	// Liquidity Providers deposit native tokens into the vault.
 	#[test]
 	fn vault_stock_dilution_k(
-		(random_index, created_accounts) in
+		(strategy_moment, accounts) in
 			valid_amounts_without_overflow_k_with_random_index(500, 1_000_000_000)
 				.prop_filter("a minimum of two accounts are required, 1 for the strategy and 1 depositor",
 							 |(_, x)| x.len() > 2)
 	) {
+		let lps_start = 1;
 		let asset_id = MockCurrencyId::D;
-		let (strategy_account_id, strategy_profits) = created_accounts[0];
-		let strategy_deposit_moment = random_index;
-		let account_start = 1;
-		let accounts = &created_accounts[account_start..];
+
+		let (strategy_account_id, strategy_deposit) = accounts[0];
+		let strategy_withdraw_pbl = Perbill::from_percent(accounts.len() as u32 % 100);
+		let strategy_withdraw = strategy_withdraw_pbl.mul_floor(strategy_deposit);
+		let strategy_vault_balance = strategy_withdraw - strategy_withdraw_pbl.mul_floor(strategy_withdraw);
```

Comment:
Instead of creating another balance on `valid_amounts_without_overflow_k_with_random_index`,  the strategy withdraw value is calculated multiplying the deposited value by `accounts.len() % 100` and then the strategy balance representing a loss is calculated using the same percentage. That said, `deposit` > `withdraw` > `strategy balance` will be truth with amounts more or less randomized.